lib/node-storage.ts

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import { StorageLike } from './local-storage';
+
+export class NodeStorage implements StorageLike {
+	private initialized = false;
+	constructor(private dataDirectory: string) {}
+
+	public async init() {
+		if (this.initialized) {
+			return;
+		}
+		this.initialized = true;
+		try {
+			await fs.mkdir(this.dataDirectory);
+		} catch {
+			// ignore if it already exists
```

Comment:
Wouldn't it make sense to surface the error?
On the other hand, specifically for `init()` the write operation would fail anyway, but what's the reasoning of also silencing the other rejections in this file?

Reply:
I think this matches what node-localstorage does, and specifically for this call the reasoning is that the error is most likely that it already exists